autoscheduler/frontend/src/tests/ui/ConfigureCard.test.tsx

```diff
@@ -26,7 +49,74 @@ describe('OptionsCard component', () => {
       fireEvent.click(getByText('Generate Schedules'));
 
       // assert
-      expect(fetch).toBeCalledTimes(1);
+      expect(fetchCount).toEqual(1);
+    });
+  });
+
+  describe('Clicking generate schedules', () => {
+    test('Does not send honors and web when "SECTION" customization level is selected', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const { getByText } = render(
+        <Provider store={store}>
+          <ConfigureCard />
+        </Provider>,
+      );
+
+      store.dispatch<any>(updateCourseCard(0, {
+        customizationLevel: CustomizationLevel.SECTION,
+        honors: 'exclude',
+        web: 'exclude',
+      }));
+
+      // Doesn't need to return anything valid
+      fetchMock.mockOnce(JSON.stringify([])); // mocks scheduler/generate call
+
+      // act
+      fireEvent.click(getByText('Generate Schedules'));
+
+      const { body } = fetchMock.mock.calls[0][1]; // Body is returned as a "blob"
+      // Convert the body into a string, parse it into an object, then get the honors & web fields
+      const { honors, web } = JSON.parse(body.toString());
+
+      // assert
+      expect(web).toBeUndefined();
+      expect(honors).toBeUndefined();
+    });
+
+    test('Does not send sections when "BASIC" customization level is selected', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const { getByText } = render(
+        <Provider store={store}>
+          <ConfigureCard />
+        </Provider>,
+      );
+
+      store.dispatch<any>(updateCourseCard(0, {
+        customizationLevel: CustomizationLevel.BASIC,
+        honors: 'exclude',
+        web: 'exclude',
+        // Add a selected section so its added to selectedSections internally
+        sections: [{
+          section: null,
+          meetings: [],
+          selected: true,
+        }],
+      }));
+
+      // Doesn't need to return anything valid
+      fetchMock.mockOnce(JSON.stringify([])); // mocks scheduler/generate call
+
+      // act
+      fireEvent.click(getByText('Generate Schedules'));
+
+      const { body } = fetchMock.mock.calls[0][1]; // Body is returned as a "blob"
+      // Convert the body into a string, parse it into an object, then get the sections field
+      const { sections } = JSON.parse(body.toString());
+
+      // assert
+      expect(sections).toBeUndefined();
```

Comment:
Based on my reading of `ConfigureCard.tsx`, shouldn't `sections` be an empty array rather than undefined?

Reply:
Yeah that's a great point. I think I ran into trouble when I tried to make it an empty array that it would be `undefined` anyways, but I'll double check and get back to you